src/vs/workbench/contrib/preferences/browser/preferences.contribution.ts

```diff
@@ -267,31 +268,89 @@ class PreferencesActionsContribution extends Disposable implements IWorkbenchCon
 				}
 			});
 		};
-		registerOpenUserSettingsEditorFromJsonAction();
-		this._register(this.userDataProfileService.onDidChangeCurrentProfile(() => registerOpenUserSettingsEditorFromJsonAction()));
 
-		registerAction2(class extends Action2 {
-			constructor() {
-				super({
-					id: SETTINGS_EDITOR_COMMAND_SWITCH_TO_JSON,
+		const openJsonFromUserSettingsDisposableStore = this._register(new DisposableStore());
+		const registerOpenJsonFromUserSettingsEditorAction = () => {
+			openJsonFromUserSettingsDisposableStore.clear();
+			if (!this.userDataProfileService.currentProfile.isDefault) {
+				const submenuId = MenuId.for('PreferencesSubMenu');
+				openJsonFromUserSettingsDisposableStore.add(registerAction2(class extends Action2 {
+					constructor() {
+						super({
+							id: SETTINGS_EDITOR_COMMAND_SWITCH_TO_JSON,
+							title: { value: nls.localize('openProfileSettings', "Open Profile Settings (JSON)"), original: 'Open Profile Settings (JSON)' },
+							f1: true,
+							menu: [{ id: submenuId }]
+						});
+					}
+					run(accessor: ServicesAccessor) {
+						const editorPane = accessor.get(IEditorService).activeEditorPane;
+						if (editorPane instanceof SettingsEditor2) {
+							return editorPane.switchToSettingsFile();
+						}
+						return null;
+					}
+				}));
+				openJsonFromUserSettingsDisposableStore.add(registerAction2(class extends Action2 {
+					constructor() {
+						super({
+							id: SETTINGS_EDITOR_COMMAND_SWITCH_TO_APPLICATION_JSON,
+							title: { value: nls.localize('openApplicationSettings', "Open Non-Profile Specific Settings (JSON)"), original: 'Open Non-Profile Specific Settings (JSON)' },
```

Comment:
Any ideas for how to make this label shorter? One suggestion was "Open Global Settings". Another one I have is "Open settings that apply to all profiles", though that's not necessarily true; it's more like "Open Default Profile Settings (JSON)".